### PR TITLE
Project log entry cleanup job

### PR DESCRIPTION
**Pull request by hennevogel**

### CODE REVIEW

File: src/api/app/models/project_log_entry.rb [hunk 1]
```diff
@@ -27,9 +27,8 @@ def self.create_from(payload, created_at, event_model_name)
     entry
   end
 
-  # Delete old entries
-  def self.clean_older_than(date)
-    where('datetime < ?', date).delete_all
+  def self.cleanup
+    where(event_type: [:build_fail, :build_success]).where('datetime < ?', Time.zone.yesterday.beginning_of_day).find_each(&:delete)
```
Comment (coolo) [inline src/api/app/models/project_log_entry.rb]: why not delete_all? There can be a lot of them after all

**Comment by coolo** [comment]
I find the concept of 'today' a little underdefined in a project with contributors in all places of the world. 

I'm already annoyed by https://metrics.opensuse.org/d/qjkd7uBmz/rails-metrics?orgId=1 defaulting to 'Today', which means basically no data available at the time I look at it. I find 'last 24 hours' a little more meaningful - no matter in what timezone you are.

File: src/api/app/models/project_log_entry.rb (hunk 1 above)
Comment (hennevogel) [reply src/api/app/models/project_log_entry.rb]: From build.o.o

```ruby
ProjectLogEntry.where(event_type: [:build_fail, :build_success]).where('datetime > ?', Time.zone.yesterday.beginning_of_day).count
=> 45580
```

And we only started yesterday at 20:00...

**Comment by hennevogel** [comment]
> I find the concept of 'today' a little underdefined in a project with contributors in all places of the world.

Good point. I change it.

File: src/api/app/models/project_log_entry.rb (hunk 1 above)
Comment (coolo) [reply src/api/app/models/project_log_entry.rb]: How does this answer my question? delete_all is one SQL statement - your code is 45581 SQL statements.

Comment (hennevogel) [reply src/api/app/models/project_log_entry.rb]: Fear of locks :cowboy_hat_face: 

https://www.youtube.com/watch?v=RwpQuvNP0R8

Comment (Ana06) [reply src/api/app/models/project_log_entry.rb]: The only problem with `delete_all` is that it goes straight to the database skipping things like callbacks and dependent rules, but we don't have anything like that in those models :+1:

File: src/api/app/models/project_log_entry.rb [hunk 2]
```diff
@@ -27,9 +27,8 @@ def self.create_from(payload, created_at, event_model_name)
     entry
   end
 
-  # Delete old entries
-  def self.clean_older_than(date)
-    where('datetime < ?', date).delete_all
+  def self.cleanup
+    where(event_type: [:build_fail, :build_success]).where('datetime < ?', Time.zone.yesterday.beginning_of_day).delete_all
```
Comment (Ana06) [inline src/api/app/models/project_log_entry.rb]: why is `beginning_of_day` needed? :thinking: `yesterday` returns an instance of the class `Date` which is already at the beginning of the day

Comment (hennevogel) [reply src/api/app/models/project_log_entry.rb]: Yeah I guess it's the same...

```
ProjectLogEntry.where(event_type: [:build_fail, :build_success]).where('datetime > ?', Time.zone.yesterday).to_sql
=> "SELECT `project_log_entries`.* FROM `project_log_entries` WHERE `project_log_entries`.`event_type` IN ('build_fail', 'build_success') AND (datetime > '2018-12-06')"
irb(main):025:0> ProjectLogEntry.where(event_type: [:build_fail, :build_success]).where('datetime > ?', Time.zone.yesterday.beginning_of_day).to_sql
=> "SELECT `project_log_entries`.* FROM `project_log_entries` WHERE `project_log_entries`.`event_type` IN ('build_fail', 'build_success') AND (datetime > '2018-12-06 00:00:00')"
```

File: src/api/app/models/project_log_entry.rb (hunk 1 above)
Comment (hennevogel) [reply src/api/app/models/project_log_entry.rb]: Yeah so does `.delete`. I was worried about it taking too long but I guess this was unfounded.

File: src/api/app/models/project_log_entry.rb (hunk 2 above)
Comment (Ana06) [reply src/api/app/models/project_log_entry.rb]: yes :eyes:

**Comment by hennevogel** [comment]
coverage my ass...

**Pull request merged**